src/main/java/com/yurjinia/auth/service/AuthService.java

```diff
@@ -43,10 +43,35 @@ public class AuthService {
     public JwtAuthenticationResponse signUp(RegistrationRequest registrationRequest, MultipartFile image) {
         userService.createUser(registrationRequest, image);
```

Comment:
user must be created when user clicked on the button in the email.

Reply:
1. before creating user you need to do the check if this user exists and confirmed email.
2. if user exists and confirmed email - throw an exception that user exists in the system.
3. if user exists and NOT confirmed email - throw an exception that user exists in the system, but didn't confirmed his email (in this case client must send the request to `/resend-confirmation` endpoint, use the example below but make it proper);
4. if user not exists - create a user and send confirmation link.

```java
@PostMapping("/resend-confirmation")
public ResponseEntity<String> resendConfirmation(@RequestParam String email) {
    User user = userService.findByEmail(email);
    if (user == null || user.isActive()) {
        return ResponseEntity.badRequest().body("Invalid request");
    }

    String token = confirmationTokenService.createToken(email);
    String confirmationLink = "http://localhost:9000/api/v1/auth/sign-up/confirm?token=" + token;
    sendConfirmationEmail(email, confirmationLink);

    return ResponseEntity.ok("Confirmation email resent");
}

```